src/node_api.cc

```diff
@@ -17,17 +16,19 @@
 #include <vector>
 #include "node_api.h"
 
-namespace v8impl {
-
-//=== Conversion between V8 Isolate and napi_env ==========================
+napi_status napi_set_last_error(napi_env env,
+                                napi_status error_code,
+                                uint32_t engine_error_code = 0,
+                                void* engine_reserved = nullptr);
+void napi_clear_last_error(napi_env env);
 
-napi_env JsEnvFromV8Isolate(v8::Isolate* isolate) {
-  return reinterpret_cast<napi_env>(isolate);
-}
+struct napi_env__ {
+  v8::Isolate* isolate;
+  v8::Persistent<v8::Value> last_exception;
```

Comment:
Will do although, since this structure is per-module, unless modules are ever unloaded, the destructor will never be called.